### DISCUSSION TITLE
How do you hand off a project to a client?

### DISCUSSION THREAD

**Discussion by ndimatteo** [vercel/vercel#5057] [Help]
**I'm having issues understanding how best to pass off a completed project to a client. The goal is:**
1. Have a project that I've dev'd on my personal, "hobby" account be managed by the client so they can be billed for usage, and be the true "owners" of the completed project.
2. Maintain access of the Github Repo so I can push updates and trigger project deploys.

# Attempts and outcomes:

### 1. "just create an account"
I initially thought the client could just create an account and I could transfer the project to them.

**Issues with this solution:**
- Creating an account requires a Github/Gitlab/Bitbucket account, which is 90% of the time, unlikely to be the case for clients.

### 2. "Create a team, and invite the client"
Then I thought, the "team" feature was the solution I was looking for since you could invite users via email and they could manage billing.

**Issues with this solution:**
- It costs $20/user, so just to have the client manage the project and give me access now costs $40/mo.
- Transfering the project requires me to disconnect my Git Integration on my "hobby" account, and reconnect at the team level. This sounds like it will block me from ever connecting my github account to my "hobby" account again for other projects I'm working on.

### 3. "Create all the accounts!"
The only other solution I can think of is:
1. Have the client provide or create a GitHub account
2. Transfer repo ownership to client GitHub account
3. Add me as a 'contributor' to transfered repo
4. Sign up for Vercel "hobby" account and setup project with Git Integration for transfered repo.

**Issues I foresee with this solution:**
- Will we lose the ability to use the initial Vercel project name/subdomain that was setup on my "hobby" account?
- If I need to access a client's vercel account, I'll need to login to multiple client accounts (seems fair, just not ideal... telling a client to pay $40/mo just so I can (more easily) access their dashboard seems like an unlikely sell)

**Comment by styfle** [comment]
Hi @ndimatteo 

Who owns the source code when a project is "complete"? You or the client?

**Reply by ndimatteo** [reply]
hey there @styfle !

If you're talking about who has _physical_ ownership of the codebase at completion... that would be me, as the repo exists on my GitHub account.

Ideally, I think it would be great to keep it that way, as often times clients do not have their own GitHub account or internal dev staff, but in the event that a client was to request that (whether at project "completion" or at a later date) I'd transfer the ownership to the client's own GitHub account.

**Reply by styfle** [reply]
It sounds like a Team would be best for this use case. You could create a GitHub Org and link to a Vercel Team. Then create a GitHub repo and link to a Vercel Project. In the Project, you can add a domain.

This gives you the most flexibility in terms of ownership because you can add and remove team members as needed, whether that is more devs on the project or handing off to the client.

Since teams are charged per user, you can avoid adding the client as a user until the project is complete.

During development, you can share the Production URL with the client or each PR's Preview URL to get feedback.

**Reply by ndimatteo** [reply]
Thanks @styfle, the issue I have with this approach is I'm still going to be the _owner_ of all my client's Vercel projects. Also creating a "Vercel Team" still only has one set of billing details associated with it, so it's not like I can create one "Team" and put all my client sites on it, as billing would need to be scoped to each client for their respective projects.

So, it would mean I'd need to create a "Team" for each client, which is what I was explaining in my original post as a solution that falls short due to the billing being a bit steep (2 users needed for each "Team") AND the Git Integration limitation of one GitHub account connected to only one Vercel team/account at a time.

Perhaps I'm missing something, but there doesn't seem to be a clear way to have my GitHub account (which houses multiple client site repos) connected to multiple client-owned and client-billed Vercel accounts.

**Reply by styfle** [reply]
> the Git Integration limitation of one GitHub account connected to only one Vercel team/account at a time.

Indeed, you would need to create a separate GitHub org for each Vercel Team to get around that limitation.

In the future, we will allow linking a single GitHub user/org to multiple Vercel Teams!

**Reply by th-km** [reply]
That'd be great because right now this workflow is totally counterproductive.

**Reply by ndimatteo** [reply]
> In the future, we will allow linking a single GitHub user/org to multiple Vercel Teams!

This is nice, but is still lacking IMO since you don't differentiate "types" of users.

I don't think it's fair to penalize developers who may be working on multiple projects and bringing in business for Vercel by charging $20 for each project they touch. Especially if the solution for client hand-off is to create teams for each client. There should be some kind of "contributor" role for developers, or vice-versa– an "accounting" role for the client who only needs to access billing.

**Reply by styfle** [reply]
We plan to add user roles in the future (tracked internally as Story 1601). However, that will probably be after the multiple team linking.

**Reply by trompx** [reply]
Hey @styfle, do you have more info about those user roles?
Same, I'd gladly bring business for Vercel but the way it is working now is preventing me from doing so.

I'm a single freelance and also wish to handle the client project code source on my github account in private repositories and have them create their Vercel account so they are billed directly.
I wish to not provide the source code to some clients (at best the code already bundled), but with the way it is working, I need to create them a github account and directly push code on their github account (so loosing the ownership of the code) as when we import a project on vercel dashboard, we need to link it directly to the project owner github account (even if I set them as collaborator).

I saw I could use the CLI and manually deploy to the client Vercel account but I'd loose the autodeploy by simply pushing updates on the github repo and I don't know how to handle multiple Vercel Ids (all of my clients acounts).
Any tips to manage this would be super helpful. Hope to be able to deploy on your platform soon!

@ndimatteo Thanks for opening this discussion Nick. Would you mind explaining how you are actually handling your client projects? Did you ended up creating teams, or do you create all your client accounts, or maybe you switched to a different static hosting provider? Would really appreciate your input.

For reference: https://github.com/vercel/vercel/discussions/5266#discussioncomment-234944

**Reply by trompx** [reply]
Hello @styfle, do you have some news concerning the possibilities to deploy client projects from other github accounts, so they are only billed and don't have access to source code?
Thanks in advance

**Reply by styfle** [reply]
> In the future, we will allow linking a single GitHub user/org to multiple Vercel Teams!

This is now possible!

However, the pricing is still the same: the team is billed for each team member.

**Reply by trompx** [reply]
So this means I can't do that with a free developper account?

**Reply by styfle** [reply]
Correct, Teams require a Pro plan.

Free accounts are for hobby projects, not commercial projects.

**Reply by tanaka-lusengo** [reply]
Hey @styfle ! 

Hope all is well. This thread was pretty informative as I'm looking to potentially start building projects for clients and would love to know if there is a recommended process for doing this on vercel? 

Thanks in advance

**Reply by styfle** [reply]
We now have separate roles: Developer, Billing, Viewer. 

https://vercel.com/docs/concepts/teams/roles-and-permissions

**Comment by zanona** [comment]
Is there currently any implication which would prevent users to create their accounts with email address instead linking their Git* accounts?

Particularly for those who want to simply deploy projects directly from the CLI.
I can't see how signing up with GitHub is advantageous? Specially because project linking is done separately?

Perhaps there're some details we're failing to understand here which would benefit from some clarification?

**Reply by styfle** [reply]
Hi @zanona 

We encourage the use of the [Git Integration](https://vercel.com/docs/git-integrations) for the optimal Vercel experience.

Without Git, you lose out on a lot of features such as faster deployments via clone, automatic git branch urls for each PR, [Branch Domains](https://vercel.com/docs/custom-domains#assigning-a-domain-to-a-git-branch), Redeploy button from the dashboard, and of course any other integrations such as Checkly, etc.

Take a look at https://vercel.com/docs/git/vercel-for-github for more info.

**Reply by zanona** [reply]
Hey, sorry for the delay on getting back to you.

Actually, I meant about the sign up process being tied to a VCS platform, in which case, it would seem more of an enforcement rather than encouragement. But I guess you guys have done some deep research about this already and may have concluded that, in order to make the best use of your platform, it's well worth filtering users from day one, so they are sure to have the right tools available, which would make sense, I guess?

Usually, my reservations are about 'relying' too much on a platform to do trivial things for me which we could effortlessly achieve with a simple shell script, for example, so we get all hyped up and forget we will possibly be giving ourselves a hard time later on, in case we ever decide to migrate, with arguments like: _“but hey, does it allow you to have different env variables per branch; does it allow you to deploy to a branch specific URL, all from the control panel?”_

But that's just me, and I know that the reality is that, yes, less piping work and more time to focus on your real project. :wink:

**Comment by cobrexus** [comment]
I totally get your points, and this is a problem I've also been trying to deal with.

If your Next.js site is *completely* static, then you can [`next build && next export`](https://nextjs.org/docs/advanced-features/static-html-export) to get static HTML which you can put on another static site host (other than Vercel).

Alternatively, the client can register the domain through a third-party (e.g. [Hover](https://www.hover.com), and you just connect the Vercel project to that domain.
